wfl/calculators/wfl_fileio_calculator.py

```diff
@@ -66,21 +66,34 @@ def setup_rundir(self):
             dir_name = str(self._cur_rundir.resolve()).replace("/", "", 1).replace("/", "_")
             directory = self._wfl_scratchdir / dir_name
             directory.mkdir(parents=True, exist_ok=True)
-            self.directory = directory
         else:
-            self.directory = self._cur_rundir
+            directory = self._cur_rundir
 
+        if self.__name__ == "Castep":
+            self._directory = directory
+        else:
+            self.directory = directory
 
     def clean_rundir(self, _default_keep_files, calculation_succeeded):
-        utils_clean_rundir(self.directory, self._wfl_keep_files, _default_keep_files, calculation_succeeded)
+
+        # Castep stores results differently ...
+        if self.__name__ == "Castep":
+            directory = self._directory
+        else:
+            directory = self.directory
+            
+        utils_clean_rundir(directory, self._wfl_keep_files, _default_keep_files, calculation_succeeded)
         if self._wfl_scratchdir is not None:
-            for f in Path(self.directory).glob("*"):
+            for f in Path(directory).glob("*"):
                 shutil.move(f, self._cur_rundir)
-            if list(Path(self.directory).iterdir()) == []:
-                warnings.warn(f"scratchdir {self.directory} is not empty, not deleting.")
+            if list(Path(directory).iterdir()) == []:
+                warnings.warn(f"scratchdir {directory} is not empty, not deleting.")
             else:
-                Path(self.directory).rmdir()
-                self.directory = '.' 
+                Path(directory).rmdir()
+                if self.__name__ == "Castep":
+                    self._direcotyr = '.'
```

Comment:
Typo.  Also, I see that this bit of code will also have to remain Castep-specific.

Reply:
Also, how important is it that we reset `self.directory`? Won't it get re-set next time, but `setup_rundir`?

Reply:
Yes, I think we can not reset `self.directory` (I don't think resetting it to `"."`  was correct anyway)